src/_nebari/provider/cloud/amazon_web_services.py

```diff
@@ -121,6 +121,30 @@ def instances(region: str) -> Dict[str, str]:
     return {t: t for t in instance_types}
 
 
+@functools.lru_cache()
+def kms_key_arns(region: str) -> Dict[str, dict]:
+    """Return dict of available/enabled KMS key IDs and associated KeyMetadata for the AWS region."""
+    session = aws_session(region=region)
+    client = session.client("kms")
+    paginator = client.get_paginator("list_keys")
+    fields = [
+        "Arn",
+        "KeyUsage",
+        "KeySpec",
+        # "KeyState",
+        # "Origin",
+        # "KeyManager",
+        # "EncryptionAlgorithms",
+        # "MultiRegion",
+    ]
```

Comment:
I would remove the extra keys from the request, and add a comment with a link to the python API docs with the extra available fields instead.

Reply:
Copy.  I'll implement this in [PR#2788](https://github.com/nebari-dev/nebari/pull/2788)